plugins/machina-tools.js

```diff
@@ -0,0 +1,554 @@
+          }
+
+          if (window.LINK_RANGE_MAC[portal.level] < child.length) {
+            lengthDescription += ' (EXCEEDS EXPECTED MAX)';
+          }
+          var childListItem = $('<li>');
+          childListItem.append(new Date(child.linkTime).toUTCString());
+          childListItem.append(' link to ');
+          var childLink = $('<a>', {
+            title: childPortal.name,
+            html: childPortal.name,
+            click: window.zoomToAndShowPortal.bind(window, child.childGuid, childPortal.latlng),
+          });
+          childListItem.append(childLink);
+          childListItem.append(`(${childPortal.level}) - ${lengthDescription}`);
+
+          childList.append(childListItem);
+        } else {
+          rc.append($('<li>', { html: `${new Date(child.linkTime).toUTCString()} link to UNKNOWN` }));
+        }
+      });
+    } else {
+      rc.append('<br/>');
+    }
+  }
+
+  return rc;
+};
+
+machinaTools.displayCluster = function (portalGuid) {
+  var clusterData = machinaTools.gatherCluster(portalGuid);
+
+  if (clusterData !== undefined) {
+    var html = $('<div>', { id: 'machina-cluster' });
+    html.append(machinaTools.clusterDisplayNode(clusterData));
+    html.append('<br/><pre>' + JSON.stringify(clusterData, null, 4) + '</pre>');
+
+    dialog({
+      html: html,
+      title: 'Machina Cluster',
+      id: 'machina-cluster',
+      width: 'auto',
+    });
+  } else {
+    dialog({
+      html: $('<div id="no-machina-cluster">No Cluster found.</div>'),
+      title: 'Machina Tools',
+      id: 'no-machina-cluster',
+    });
+  }
+};
+
+function createInfoLink(text, title, clickCallback) {
+  var aside = $('<aside>');
+  $('<a>', { title: title, click: clickCallback, html: text }).appendTo(aside);
+  return aside;
+}
+
+machinaTools.onPortalDetailsUpdated = function () {
+  var portalData;
+
+  // If the portal was cleared then exit.
+  if (window.selectedPortal === null) return;
+
+  portalData = portalDetail.get(window.selectedPortal);
+
+  if (portalData.team === 'M') {
+    var linkdetails = $('.linkdetails');
+    linkdetails.append(createInfoLink('Find Parent', 'Find Machina Parent', () => window.plugin.machinaTools.goToParent(window.selectedPortal)));
+    linkdetails.append(createInfoLink('Find Seed', 'Find Machina Seed', () => window.plugin.machinaTools.goToSeed(window.selectedPortal)));
+    linkdetails.append(createInfoLink('Cluster Details', 'Display Machina Cluster', () => window.plugin.machinaTools.displayCluster(window.selectedPortal)));
+
+    // Add this portal's conflict zone to the conflict area
+    machinaTools.drawPortalExclusion(window.selectedPortal);
+  }
+};
+
+/**
+ * Indicates whether portals are displayed at the current level.  Simply using zoom level
+ * does not factor in other tools that adjust display capabilities.
+ */
+machinaTools.zoomLevelHasPortals = function () {
+  return window.getMapZoomTileParameters(window.getDataZoomForMapZoom(window.map.getZoom())).hasPortals;
```

Comment:
short-hand used in other plugins

```suggestion
  return window.getDataZoomTileParameters().hasPortals;
```